app/models/comment_observer.rb

```diff
@@ -132,6 +103,80 @@ def after_update(comment)
   end
 
   protected
+  
+    def notify_parent_comment_owner(comment)
+      if comment.reply_comment? && !comment.unreviewed?
+        parent_comment = comment.commentable
+        parent_comment_owner = parent_comment.comment_owner # will be nil if not a user, including if an admin
+
+        # if I'm replying to a comment you left for me, mark your comment as replied to in my inbox
+        if comment.comment_owner
+          if (inbox_comment = comment.comment_owner.inbox_comments.find_by_feedback_comment_id(parent_comment.id))
+            inbox_comment.update_attributes(:replied_to => true, :read => true)
+          end
+        end
+        
+        # send notification to the owner of the original comment if they're not the same as the commenter
+        if (have_different_parent_owner?(comment, parent_comment)) 
+          if !parent_comment_owner || notify_user_by_email?(parent_comment_owner) || comment.ultimate_parent.is_a?(Tag)
+            if comment.edited_at_changed?
+              CommentMailer.edited_comment_reply_notification(parent_comment.id, comment.id).deliver
+            else
+              CommentMailer.comment_reply_notification(parent_comment.id, comment.id).deliver
+            end
+          end
+          if parent_comment_owner && notify_user_by_inbox?(parent_comment_owner)
+            if comment.edited_at_changed?
+              update_feedback_in_inbox(parent_comment_owner, comment)
+            else
+              add_feedback_to_inbox(parent_comment_owner, comment)
+            end
+          end
+          if parent_comment_owner
+            return parent_comment_owner
+          end
+        end
+      end
+      return nil
+    end
+    
+    def have_different_parent_owner?(comment, parent_comment)
+      parent_comment_owner = parent_comment.comment_owner # will be nil if not a user, including if an admin
+      return (!parent_comment_owner && parent_comment.comment_owner_email && parent_comment.comment_owner_name) ||
+                  (parent_comment_owner && (parent_comment_owner != comment.comment_owner))
+    end
+    
+    def content_too_different?(new_content, old_content)
+      # we added more than the threshold # of chars, just return
+      return true if new_content.length > (old_content.length + ArchiveConfig.COMMENT_MODERATION_THRESHOLD)
+
+      # quick and dirty iteration to compare the two strings 
+      cost = 0
+      new_i = 0
+      old_i = 0
+      while new_i < new_content.length && old_i < old_content.length
+        if new_content[new_i] == old_content[old_i]
+          new_i += 1; old_i +=1; next
```

Comment:
Do not use semicolons to terminate expressions.<br>Surrounding space missing for operator `+=`.